### ISSUE TITLE
Upstream error fix | Validate that birthdates values have leading zeros

### ISSUE THREAD

**Issue by leikkisa** [department-of-veterans-affairs/caseflow#13589]
## Description
When a veteran's birthdate's day was "5" instead of "05", it gave us an upstream error.

## Acceptance criteria
- [ ] Validate that a veteran's BGS record field "birth_day" has leading zeros before starting an intake
- [ ] If it does not, return an error to the user explaining that they need to update the birthdate
- [ ] Look at some veteran BGS records to see if months also have leading zeros, if so, apply the same validation above
- [ ] Update documentation: Create new document about intake validation

## Background/context/resources
Here was the upstream error ticket:
https://github.com/department-of-veterans-affairs/caseflow/issues/12590

## Technical notes
To get a veteran's bgs_record, you can call veteran.bgs_record

Check out the DecisionReviewIntake model as a starting place to look at validation.

**Comment by leikkisa** [comment]
@rutvigupta-design Do you have suggestions for the copy on this?

**Comment by rutvigupta-design** [comment]
For now, I think we can proceed without including which system this should be updated in since as far as I know, we don't include that in most errors. I am assuming that the year is 'xxxx' - is this correct? 

![Veteran Birthdate Error](https://user-images.githubusercontent.com/51714306/77468125-2d177980-6ddb-11ea-9426-f80e2f2a4c23.jpg)

**Comment by leikkisa** [comment]
@rutvigupta-design I don't how the user enters birthdates in other UIs, like if it's one string or multiple fields. Here's an example of the data we see:
```
birth_month=>"12", :birth_day=>"05", :birth_century=>"19", :birth_year=>"60", :date_of_birth=>"12/05/1960"
```

The issue we had is that the day was missing a leading zero.

For the copy:
I'm more used to seeing "mm/dd/yyyy" which makes it clear which is the month and the day.  It might be valuable to include "(including leading zeros)" after the format. Not sure if that's necessary.

I don't think "to create an EP" is necessary in the title, and I like the idea of adding the systems to check too like you're doing elsewhere. Jane got back to us and said VBMS or Share.

**Comment by rutvigupta-design** [comment]
@leikkisa totally agreed with the "mm/dd/yyyy" format - thanks for pointing that out! I think leading zeros is a very technical term, so I just added (please include zeros) and removed the example, since it was a little redundant and the format hopefully makes it clear enough. 

![Veteran Birthdate Error](https://user-images.githubusercontent.com/51714306/77479179-4c1f0700-6ded-11ea-95d1-aed674ce26c9.jpg)

**Comment by leikkisa** [comment]
@rutvigupta-design Nice suggestion on "please include zeros". This looks great to me.

**Comment by Sjones352** [comment]
@rutvigupta-design and @leikkisa I think it would be wise to keep the same message title as the rest of all upstream errors we have in intake. This is because it pertains to the same thing as veteran having invalid information to create an EP. In the screenshot below, it shows the message that is clearly missing to continue with an intake. 

<img width="969" alt="Screen Shot 2020-03-28 at 3 20 20 AM" src="https://user-images.githubusercontent.com/23080951/77817662-1b67f780-70a3-11ea-9f9b-7ce2343738d8.png">

**Comment by rutvigupta-design** [comment]
@Sjones352 great point about the 3 please's, you're right, it's redundant! While I think consistency is great, I think the upstream error is a little vague and would love to work towards making all errors action-oriented. What do you think of this? 
![Birthdate Error](https://user-images.githubusercontent.com/51714306/77925363-62ecb000-726a-11ea-8fc5-8926be55471b.jpg)

**Comment by leikkisa** [comment]
@Sjones352 Hi Sandra - I am a fan of the error message above. Small comment, Share is not in all caps. I used to think so, but people have mostly spelled it with just the "S" capitalized. Thanks!

**Comment by Sjones352** [comment]
@rutvigupta-design and @leikkisa would this title copy change apply to the `veteranAddressInvalid`, veteranCityInvalidFields, and veteranAddressTooLong since this is on the same review page, that I am working on.

**Comment by rutvigupta-design** [comment]
@Sjones352 as per our conversation, we will have "Check the Veteran's profile for invalid information" as the title for all upstream errors. cc @leikkisa 
Low priority - over time, auditing error messages that exist in Intake to check for consistency to eventually create a styleguide for Intake error messages.
